packages/headless/src/controllers/commerce/core/facets/category/headless-commerce-category-facet-search.ts

```diff
@@ -37,11 +38,19 @@ export function buildCategoryFacetSearch(
   }
 
   const {showMoreResults, updateCaptions, ...restOfFacetSearch} =
-    buildCoreCategoryFacetSearch(engine, {
-      ...props,
-      executeFacetSearchActionCreator: executeCommerceFacetSearch,
-      executeFieldSuggestActionCreator: executeCommerceFieldSuggest,
-    });
+    buildCoreCategoryFacetSearch(
+      {
+        ...engine,
+        get state() {
+          return engine[stateKey];
+        },
+      },
```

Comment:
Why do we need to do this? Won't the `buildCoreCategoryFacetSearch` access the engine state through the `stateKey` as well?

Reply:
Initially I hesitated to change the type of the "engine" argument to CoreEngine<...> | CoreEngineNext<...>, but I did so, so oddness is now gone 😄